src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -1536,123 +1536,158 @@ private String[] SplitInternal(String separator, String[] separators, Int32 coun
+        {
+            if (null == trimChars || trimChars.Length == 0)
+            {
                 return TrimHelper(TrimHead);
             }
-            return TrimHelper(trimChars,TrimHead);
+            fixed (char* pTrimChars = trimChars)
+            {
+                return TrimHelper(pTrimChars, trimChars.Length, TrimHead);
+            }
         }
 
-        // Removes a set of characters from the beginning of this string.
-        public String TrimStart() => TrimHelper(TrimHead);
-        
+        // Removes a set of characters from the end of this string.
+        public string TrimEnd() => TrimHelper(TrimTail);
+
+        // Removes a set of characters from the end of this string.
+        public unsafe string TrimEnd(char trimChar) => TrimHelper(&trimChar, 1, TrimTail);
 
         // Removes a set of characters from the end of this string.
-        public String TrimEnd(params char[] trimChars) {
-            if (null==trimChars || trimChars.Length == 0) {
+        public unsafe string TrimEnd(params char[] trimChars)
+        {
+            if (null == trimChars || trimChars.Length == 0)
+            {
                 return TrimHelper(TrimTail);
             }
-            return TrimHelper(trimChars,TrimTail);
+            fixed (char* pTrimChars = trimChars)
+            {
+                return TrimHelper(pTrimChars, trimChars.Length, TrimTail);
+            }
         }
-        
-        // Removes a set of characters from the end of this string.
-        public String TrimEnd()=> TrimHelper(TrimTail);
-        
-
-        // Trims the whitespace from both ends of the string.  Whitespace is defined by
-        // Char.IsWhiteSpace.
-        //
-        [Pure]
-        public String Trim() {
-            Contract.Ensures(Contract.Result<String>() != null);
-            Contract.EndContractBlock();
 
-            return TrimHelper(TrimBoth);        
-        }
+        private string TrimHelper(int trimType)
+        {
+            // end will point to the first non-trimmed character on the right.
+            // start will point to the first non-trimmed character on the left.
+            int end = Length - 1;
+            int start = 0;
 
-       
-        private String TrimHelper(int trimType) {
-            //end will point to the first non-trimmed character on the right
-            //start will point to the first non-trimmed character on the Left
-            int end = this.Length-1;
-            int start=0;
-
-            //Trim specified characters.
-            if (trimType !=TrimTail)  {
-                for (start=0; start < this.Length; start++) {
-                    if (!Char.IsWhiteSpace(this[start])) break;
+            // Trim specified characters.
+            if (trimType != TrimTail)
+            {
+                for (start = 0; start < Length; start++)
+                {
+                    if (!char.IsWhiteSpace(this[start])) break;
```

Comment:
For what its worth, the [coding guidelines](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md) allow single line if without braces. Anything else must have braces.

That isn't how I was brought up, but those are the rules :)